subsys/net/lib/ptp/port.c

```diff
@@ -89,8 +89,8 @@ static int port_msg_send(struct ptp_port *port, struct ptp_msg *msg, enum ptp_so
 static void port_timer_set_timeout(struct k_timer *timer, uint8_t factor, int8_t log_seconds)
 {
 	uint64_t timeout = log_seconds < 0 ?
-		(uint64_t)((NSEC_PER_SEC * factor) >> (log_seconds * -1)) :
-		(uint64_t)((NSEC_PER_SEC * factor) << log_seconds);
+		(uint64_t)(NSEC_PER_SEC * factor) >> -log_seconds :
```

Comment:
Wouldn't the overflow still be the case though? If we cast the multiplication result only, the multiplication would still be done with 32-bit arithmetic which may overflow. Shouldn't we cast one of the multiplication factors to 64-bit so that 64-bit arithmetic is used for multiplication? Something like
` ((uint64_t)NSEC_PER_SEC * factor) >> -log_seconds`

Reply:
You're right in my commit multiplication is still done on unsigned integers. Let me fix that!